### PR TITLE
Better error for missing Vault config when using credentials provider

### PR DESCRIPTION
**Pull request by kdubb**
Replaces an NPE with a better exception & message if you have a missing Vault configuration when you are trying to use the Vault credentials provider.

### CODE REVIEW

**Comment by kdubb** [comment]
@gsmet I promise this is the last Vault change in my queue 😉

**Review by gsmet — CHANGES_REQUESTED** [review]
Let's not merge this right away. I'm still waiting for the final approval for the extension move.

If I get it, the PR will have to be made on another repo.

**Comment by gsmet** [comment]
I reported this PR here: https://github.com/quarkiverse/quarkus-vault/pull/1 with proper authorship.

**Pull request closed without merging**
